pkg/source/reconciler/mtsource/controller.go

```diff
@@ -119,15 +120,19 @@ func NewController(
 
 		patched, err := kafkaclient.Get(ctx).SourcesV1beta1().KafkaSources(key.Namespace).Patch(ctx, key.Name, types.JSONPatchType, patch, metav1.PatchOptions{}, "status")
 		if err != nil {
-			return fmt.Errorf("Failed patching: %w", err)
+			return fmt.Errorf("failed patching: %w", err)
 		}
 		logging.FromContext(ctx).Debugw("Patched resource", zap.Any("patch", patch), zap.Any("patched", patched))
 		return nil
 	}
 
-	c.scheduler = stsscheduler.NewScheduler(ctx,
-		system.Namespace(), mtadapterName, c.vpodLister, rp, env.PodCapacity, env.SchedulerPolicy,
-		nodeInformer.Lister(), evictor)
+	policy := &stsscheduler.SchedulerPolicy{}
+	if err := initPolicyFromConfigMap(ctx, env.SchedulerPolicyConfigMap, policy); err != nil {
+		return nil
```

Comment:
policies are immutable, admin-controlled parameters so it's fine to `panic` here.